crates/interpreter/src/util.rs

```diff
@@ -12,10 +12,124 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::error::*;
+
 // TODO(dev/fast-interp): Add debug asserts when `off` is positive and negative, and `toctou`
 // support.
 pub fn offset_front<T>(cur: &[T], off: isize) -> &[T] {
     unsafe {
         core::slice::from_raw_parts(cur.as_ptr().offset(off), (cur.len() as isize - off) as usize)
     }
 }
+
+#[allow(dead_code)]
+pub fn offset_front_check<'a, T>(beg: &'a [T], cur: &'a [T], off: isize) -> Result<&'a [T], Error> {
+    if beg.is_empty() || cur.is_empty() {
+        #[cfg(feature = "debug")]
+        eprintln!("One or both slices are empty.");
```

Comment:
This should not be an error. It should be valid to call `offset_front_check(xs, xs, 0)` where `xs.is_empty()`.